cpp/tests/stream_compaction/distinct_tests.cpp

```diff
@@ -28,218 +28,506 @@
 #include <cudf/table/table_view.hpp>
 #include <cudf/types.hpp>
 
-#include <algorithm>
 #include <cmath>
 
+auto constexpr null{0};  // null at current level
+auto constexpr XXX{0};   // null pushed down from parent level
+auto constexpr KEEP_ANY   = cudf::duplicate_keep_option::KEEP_ANY;
+auto constexpr KEEP_FIRST = cudf::duplicate_keep_option::KEEP_FIRST;
+auto constexpr KEEP_LAST  = cudf::duplicate_keep_option::KEEP_LAST;
+auto constexpr KEEP_NONE  = cudf::duplicate_keep_option::KEEP_NONE;
+
+using int32s_col  = cudf::test::fixed_width_column_wrapper<int32_t>;
+using floats_col  = cudf::test::fixed_width_column_wrapper<float>;
+using lists_col   = cudf::test::lists_column_wrapper<int32_t>;
+using strings_col = cudf::test::strings_column_wrapper;
+using structs_col = cudf::test::structs_column_wrapper;
+
 using cudf::nan_policy;
 using cudf::null_equality;
 using cudf::null_policy;
+using cudf::test::iterators::no_nulls;
+using cudf::test::iterators::null_at;
+using cudf::test::iterators::nulls_at;
 
 struct Distinct : public cudf::test::BaseFixture {
 };
 
-TEST_F(Distinct, StringKeyColumn)
+TEST_F(Distinct, StringKeyColumn_KEEP_ANY)
```

Comment:
We can leave the existing test as it is now but renames the test fixture as `DistinctKeepAny`. The suffix like `_KEEP_ANY` or `_KEEP_EXCEPT_ANY` is odd.